TweetMate/src/main/java/com/ybsystem/tweetmate/fragments/dialog/TweetDialog.java

```diff
@@ -0,0 +1,352 @@
+
+        // Create dialog
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        builder.setView(view);
+
+        return builder.create();
+    }
+
+    @Override
+    public void onDismiss(DialogInterface dialog) {
+        dismiss();
+    }
+
+    private void renderTweet(TweetRow tweetRow) {
+        // Set all fields
+        tweetRow.initVisibilities();
+        tweetRow.setStatus(mStatus);
+        tweetRow.setUserName();
+        tweetRow.setScreenName();
+        tweetRow.setUserIcon();
+        tweetRow.setTweetText();
+        tweetRow.setRelativeTime();
+        tweetRow.setAbsoluteTime();
+        tweetRow.setVia();
+        tweetRow.setRtFavCount();
+        tweetRow.setRetweetedBy();
+        tweetRow.setMarks();
+        tweetRow.setThumbnail();
+        tweetRow.setQuoteTweet();
+        tweetRow.setBackgroundColor();
+    }
+
+    @Data
+    private static class Menu {
+        private int drawable;
+        private int color;
+        private String label;
+        private String userIconURL;
+        private Runnable runnable;
+
+        public Menu(int drawable, int color, String label,
+                    String userIconURL, Runnable runnable) {
+            this.drawable = drawable;
+            this.color = color;
+            this.label = label;
+            this.userIconURL = userIconURL;
+            this.runnable = runnable;
+        }
+    }
+
+    private static class MenuArrayAdapter extends ArrayAdapter<Menu> {
+
+        public MenuArrayAdapter(Context context) {
+            super(context, android.R.layout.simple_list_item_1);
+        }
+
+        @Override
+        public View getView(final int position, View convertView, ViewGroup parent) {
+            // Inflate
+            View cv = convertView;
+            if (cv == null) {
+                cv = LayoutInflater.from(getContext())
+                        .inflate(R.layout.list_item_tweet_dialog, parent, false);
+            }
+            // Get item
+            Menu menu = getItem(position);
+
+            // Set image
+            ImageView icon = cv.findViewById(R.id.image_icon);
+            ImageView userIcon = cv.findViewById(R.id.image_user_icon);
+            if (menu.getUserIconURL().equals("")) {
+                // Icon
+                icon.setImageResource(menu.getDrawable());
+                icon.setColorFilter(menu.getColor());
+                icon.setVisibility(View.VISIBLE);
+                userIcon.setVisibility(View.GONE);
+            } else {
+                // UserIcon
+                GlideUtils.load(menu.getUserIconURL(), userIcon, CIRCLE);
+                userIcon.setVisibility(View.VISIBLE);
+                icon.setVisibility(View.GONE);
+            }
+
+            // Set text
+            TextView tv = cv.findViewById(R.id.text_item);
+            tv.setText(menu.getLabel());
+
+            return cv;
+        }
+    }
+
+    private void setMenuItems(MenuArrayAdapter adapter) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 38 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS3776)